proxy/backend/src/main/java/org/apache/shardingsphere/proxy/backend/response/header/update/UpdateResponseHeader.java

```diff
@@ -43,24 +44,40 @@ public final class UpdateResponseHeader implements ResponseHeader {
     private long updateCount;
     
     public UpdateResponseHeader(final SQLStatement sqlStatement) {
-        this(sqlStatement, Collections.emptyList());
+        this(sqlStatement, Collections.emptyList(), Collections.emptyList());
     }
     
     public UpdateResponseHeader(final SQLStatement sqlStatement, final Collection<UpdateResult> updateResults) {
+        this(sqlStatement, updateResults, Collections.emptyList());
+    }
+    
+    public UpdateResponseHeader(final SQLStatement sqlStatement, final Collection<UpdateResult> updateResults, final Collection<Comparable<?>> autoIncrementGeneratedValues) {
         this.sqlStatement = sqlStatement;
-        lastInsertId = getLastInsertId(updateResults);
+        lastInsertId = getLastInsertId(updateResults, autoIncrementGeneratedValues);
         updateCount = updateResults.iterator().hasNext() ? updateResults.iterator().next().getUpdateCount() : 0;
         for (UpdateResult each : updateResults) {
             updateCounts.add(each.getUpdateCount());
         }
     }
     
-    private long getLastInsertId(final Collection<UpdateResult> updateResults) {
-        long result = 0;
+    private long getLastInsertId(final Collection<UpdateResult> updateResults, final Collection<Comparable<?>> autoIncrementGeneratedValues) {
+        List<Long> lastInsertIds = new LinkedList<>();
```

Comment:
It's better to use `ArrayList` here if we are going to sort it. Or could we just keep a minimum long value in variable?

Reply:
Thank you for your suggestion, I will optimize it.